### PR TITLE
IHospitalVisitApi - Refactor away from IApiResponse<T>

### PR DESCRIPTION
**Pull request by rayking29**
# Fixes [AB#14339](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14339)

## Description

IHospitalVisitApi - Refactor away from IApiResponse<T>
- removed IApiResponse and refactored code
- added and removed unit tests for refactored code
- changed method names to reflect async calls

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

Screenshot 2022-11-22 at 9 31 00 AM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs [hunk 1]
```diff
@@ -91,71 +92,25 @@ public async Task<RequestResult<PhsaResult<IEnumerable<HospitalVisit>>>> GetHosp
...
             {
-                this.logger.LogCritical("HTTP Request Exception {Error}", e.ToString());
+                this.logger.LogError("Error while retrieving Hospital Visits... {Error}", e);
+                HttpStatusCode? statusCode = (e as ApiException)?.StatusCode ?? ((HttpRequestException)e).StatusCode;
                 requestResult.ResultError = new()
                 {
-                    ResultMessage = "Error with HTTP Request",
+                    ResultMessage = $"Status: {statusCode}. Error while retrieving Hospital Visits",
                     ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Phsa),
                 };
```
Comment (MikeLyttle) [inline Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs:109]: We should discuss if we want to include specific information (e.g. status code) in the the result message moving forward.

Comment (sslaws) [reply Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs:109]: I don't think we should

File: Apps/Encounter/test/unit/Delegates/RestHospitalVisitDelegateTests.cs [hunk 2]
```diff
@@ -109,97 +90,65 @@ public void ShouldGetHospitalVisits()
...
 
-            return hospitalVisitDelegate;
+            // Act
+            RequestResult<PhsaResult<IEnumerable<HospitalVisit>>> actualResult = hospitalVisitDelegate.GetHospitalVisitsAsync(It.IsAny<string>()).Result;
+
+            // Assert
+            Assert.Equal(ResultType.Error, actualResult.ResultStatus);
+            Assert.Equal(expectedMessage, actualResult.ResultError?.ResultMessage);
+            Assert.True(actualResult.TotalResultCount == 0);
+            Assert.True(actualResult.PageSize == int.Parse(ConfigFetchSize, CultureInfo.InvariantCulture));
```
[the remark this answers is not in the document]
Comment (rayking29) [reply Apps/Encounter/test/unit/Delegates/RestHospitalVisitDelegateTests.cs:151]: Fixed

**Review by BrianMaki — DISMISSED** [review]

File: Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs (hunk 1 above)
Comment (rayking29) [reply Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs:109]: Okay I can change that. I was using the example from RestMspVisitDelegate in Encounter, should I change it there too?

Comment (BrianMaki) [reply Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs:109]: The message should at least differentiate the ApiException from the HttpRequestException.  Having a code or a more definitive message is better than a generic one where you will be forced to search through Kibana for why this occurred.

Moreover, these exceptions were rarely occur.  These are not validation errors.

**Review by sslaws — DISMISSED** [review]

**Review by MikeLyttle — APPROVED** [review]

File: Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs]: Also the status code is retrieved from the ApiException and HttpRequestException.  No problems with a generic message.

**Review by sslaws — APPROVED** [review]

**Pull request merged**